app/models/user.rb

```diff
@@ -12,6 +12,10 @@ class User < ApplicationRecord
   has_many :comments, foreign_key: :author_id
   has_many :likes, foreign_key: :author_id
 
```

Comment:
- [ OPTIONAL] 
You could create user.admin? method in the user model and call it in the ability class. This is a better way, but nevertheless, your code works which is good.

_app/models/user.rb_
```ruby
   def admin?
     role == 'admin' ? true : false
   end
```

_app/models/ability.rb_
```ruby
     def initialize(user)

         user ||= User.new # guest user (not logged in)

         if user.admin? 
         can :manage, :all
         else
         can :manage, Post, author: user  # owner of post can perform all view, update or destroy own post if Post.author == user
         can :manage, Comment, author: user
         can :read, :all
         end
     end
```